pending-documentation.md

```diff
@@ -107,6 +107,143 @@ argument and parameter are K, and it has no effect on the resulting sort of
 the term. However, it will nevertheless affect the kore generated from the term
 by introducing an additional parameter to the symbol generated for the term.
 
+### `strict` and `seqstrict` attributes
+
+The strictness attributes allow defining evaluation strategies without having to
+explicitely make rules which implement them. This is done by injecting *heating*
+and *cooling* rules for the subterms. For this to work, you need to define what
+a *result* is for K, by extending the  `KResult` sort.
+
+For example:
+
+```k
+syntax AExp ::= Int
+              | AExp "+" AExp [strict]
+```
+
+This generates two heating rules (where the hole syntax `[]` is automatically
+added to create an evaluation context):
+
+```k
+rule <k> AE1:AExp + AE2:AExp => AE1 ~> [] + AE2 ... </k> requires notBool isKResult(AE1)
+rule <k> AE1:AExp + AE2:AExp => AE2 ~> AE1 + [] ... </k> requires notBool isKResult(AE2)
+```
+
+And two corresponding cooling rules:
+
+```k
+rule <k> AE1:KResult ~> [] + AE2 => AE1 + AE2 ... </k>
+rule <k> AE2:KResult ~> AE1 + [] => AE1 + AE2 ... </k>
+```
+
+Note that these rules depend on the definition of `KResult`, so you need to
+define this for K. For instance, we tell K that a term is fully evaluated once it
+becomes an `Int` here:
+
+```k
+syntax KResult ::= Int
+```
+
+Note that you can also say that a given expression is only strict only in
+specific argument positions. Here we use this to define "short-circuiting"
+boolean operators.
+
+```k
+syntax KResult ::= Bool
+
+syntax BExp ::= Bool
+              | BExp "||" BExp [strict(1)]
+              | BExp "&&" BExp [strict(1)]
+
+rule <k> true  || _    => true ... </k>
+rule <k> false || REST => REST ... </k>
+
+rule <k> true  && REST => REST  ... </k>
+rule <k> false && _    => false ... </k>
+```
+
+If you want to force a specific evaluation order of the arguments, you can use
+the variant `seqstrict` to do so. For example, this would make the boolean
+operators short-circuit in their _second_ argument first:
+
+```k
+syntax KResult ::= Bool
+
+syntax BExp ::= Bool
+              | BExp "||" BExp [seqstrict(2,1)]
+              | BExp "&&" BExp [seqstrict(2,1)]
+
+rule <k> _    || true  => true ... </k>
+rule <k> REST || false => REST ... </k>
+
+rule <k> REST && true  => REST  ... </k>
+rule <k> _    && false => false ... </k>
```

Comment:
missing from this discussion are contexts, context aliases, the `heat` and `cool` attributes, and the `hybrid` attribute.

Reply:
I won't add them.